catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/util/UiHardwareTestUtils.java

```diff
@@ -0,0 +1,142 @@
+import org.catrobat.catroid.ui.adapter.NfcTagAdapter;
+import org.catrobat.catroid.ui.fragment.NfcTagFragment;
+
+import java.util.List;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils.onScriptList;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.SpinnerUtils.checkIfSpinnerOnBrickAtPositionShowsString;
+import static org.catrobat.catroid.uiespresso.util.UiTestUtils.getCurrentActivity;
+import static org.catrobat.catroid.uiespresso.util.UiTestUtils.getResourcesString;
+
+public final class UiHardwareTestUtils {
+	// Suppress default constructor for non-instantiability
+	private UiHardwareTestUtils() {
+		throw new AssertionError();
+	}
+
+	public static void fakeNfcTag(String uid, NdefMessage ndefMessage, Tag tag) {
+		Class activityCls = UiTestUtils.getCurrentActivity().getClass();
+		Context packageContext = UiTestUtils.getCurrentActivity();
+
+		PendingIntent pendingIntent = PendingIntent.getActivity(packageContext, 0,
+				new Intent(packageContext, activityCls).addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP), 0);
+
+		String intentAction = NfcAdapter.ACTION_TAG_DISCOVERED;
+		byte[] tagId = uid.getBytes();
+
+		Intent intent = new Intent();
+		intent.setAction(intentAction);
+		if (tag != null) {
+			intent.putExtra(NfcAdapter.EXTRA_TAG, tag);
+		}
+		intent.putExtra(NfcAdapter.EXTRA_ID, tagId);
+		if (ndefMessage != null) {
+			intent.putExtra(NfcAdapter.EXTRA_NDEF_MESSAGES, new NdefMessage[] {ndefMessage});
+			// todo: Why do we need this check?
+			if (NfcAdapter.ACTION_NDEF_DISCOVERED.equals(intentAction)) {
+				Uri uri = ndefMessage.getRecords()[0].toUri();
+				String mime = ndefMessage.getRecords()[0].toMimeType();
+				if (uri != null) {
+					intent.setData(uri);
+				} else {
+					intent.setType(mime);
+				}
+			}
+		}
+
+		try {
+			pendingIntent.send(packageContext, Activity.RESULT_OK, intent);
+		} catch (PendingIntent.CanceledException e) {
+			Log.d("fakeNfcTag", e.getMessage());
+		}
+	}
+
+	public static void enableNfcBricks(Context context) {
+		PreferenceManager.getDefaultSharedPreferences(context)
+				.edit()
+					.putBoolean(SettingsActivity.SETTINGS_SHOW_NFC_BRICKS, true)
+				.commit();
+	}
+
+	public static void gotoNfcFragment(int nfcBrickPosition) {
+		onScriptList().atPosition(nfcBrickPosition).onChildView(withId(R.id.brick_when_nfc_spinner))
+				.perform(click());
+		onView(withText(R.string.new_nfc_tag)).perform(click());
```

Comment:
code style, please have all espresso actions

.perform
.check

in a new line